From e12839e74a97ea72ada5e6eaa0fae53866a6e0d3 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Tue, 17 Oct 2017 23:39:27 +0200
Subject: [PATCH 3/8] mediaserver: Allow finding the hal_camera hardware
 service

 * denied { find } for
    interface=android.hardware.camera.provider::ICameraProvider
    scontext=u:r:mediaserver:s0
    tcontext=u:object_r:hal_camera_hwservice:s0 tclass=hwservice_manager

Change-Id: I5f61a5879f74ea4f1e29246784af512fea8f8127
---
 public/mediaserver.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/public/mediaserver.te b/public/mediaserver.te
index 074c8f6..9472a44 100644
--- a/public/mediaserver.te
+++ b/public/mediaserver.te
@@ -106,6 +106,7 @@ allow mediaserver hidl_token_hwservice:hwservice_manager find;
 ifelse(target_has_legacy_camera_hal1, `true',
   allow mediaserver cameraproxy_service:service_manager find;
   allow mediaserver cameraserver_service:service_manager add;
+  allow mediaserver hal_camera_hwservice:hwservice_manager find;
 ,
 )
 
-- 
2.7.4

